BigBaseV2/src/views/network/view_session.cpp

```diff
@@ -1,17 +1,64 @@
 #include "fiber_pool.hpp"
 #include "util/session.hpp"
 #include "views/view.hpp"
+#include "pointers.hpp"
+#include "gta/joaat.hpp"
+#include "util/notify.hpp"
+#include "rage/rlSessionByGamerTaskResult.hpp"
 
 namespace big
 {
+	static void join_by_rockstar_id(uint64_t rid)
+	{
+		if (SCRIPT::GET_NUMBER_OF_THREADS_RUNNING_THE_SCRIPT_WITH_THIS_HASH(RAGE_JOAAT("maintransition")) != 0 ||
+			STREAMING::IS_PLAYER_SWITCH_IN_PROGRESS())
+		{
+			notify::above_map("Cannot RID join now");
+			return;
+		}
+
+		rage::rlGamerHandle player_handle(rid);
+		rage::rlSessionByGamerTaskResult result;
+		bool success = false;
+		int state = 0;
+		if (g_pointers->m_start_get_session_by_gamer_handle(0, &player_handle, 1, &result, 1, &success, &state))
+		{
+			while (state == 1)
+				script::get_current()->yield();
+
+			if (state == 3 && success)
+			{
+				g->session.join_queued = true;
+				g->session.info = result.m_session_info;
+				big::session::join_type({ eSessionType::NEW_PUBLIC });
+				if (SCRIPT::GET_NUMBER_OF_THREADS_RUNNING_THE_SCRIPT_WITH_THIS_HASH(RAGE_JOAAT("maintransition")) == 0)
+				{
+					g->session.join_queued = false;
+					notify::above_map("RID join failed, unable to launch maintransition");
```

Comment:
Might get obstructed by the player list rendering over this, prefer using the notification service.